GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,619 @@
+
+### Proposal categories
+
+There are two categories of proposal:
+- `Regular`
+- `Urgent`
+
+These two categories are strictly identical except that `Urgent` proposals can be accepted faster if a certain condition is met. For more information, see **Thresholds** section.
+
+
+## Vote
+
+### Participants
+
+*Participants* are users that have the right to vote. On the Cosmos Hub, participants are bonded Atom holders. Unbonded Atom holders and other users do not get the right to participate in governance. However, they can submit and deposit on proposals.
+
+### Voting period
+
+Once a proposal reaches `MinDeposit`, it immediately enters `Voting period`. We define `Voting period` as the interval between the moment the vote opens and the moment the vote closes. `Voting period` should always be shorter than `Unbonding period` to prevent double voting. The initial value of `Voting period` is 2 weeks.
+
+### Option set
+
+The option set of a proposal refers to the set of choices a participant can choose from when casting its vote.
+
+The initial option set includes the following options: 
+- `Yes`
+- `No`
+- `NoWithVeto` 
+- `Abstain` 
+
+`NoWithVeto` counts as `No` but also adds a `Veto` vote. `Abstain` option allows voters to signal that they do not intend to vote in favor or against the proposal but accept the result of the vote. 
+
+*Note: from the UI, for urgent proposals we should maybe add a ‘Not Urgent’ option that casts a `NoWithVeto` vote.*
+
+### Quorum 
+
+Quorum is defined as the minimum percentage of voting power that needs to be casted on a proposal for the result to be valid. 
+
+In the initial version of the governance module, there will be no quorum enforced by the protocol. Participation is ensured via the combination of inheritance and validator's punishment for non-voting.
+
+### Threshold
+
+Threshold is defined as the minimum ratio of `Yes` votes to `No` votes for the proposal to be accepted.
+
+Initially, the threshold is set at 50% with a possibility to veto if more than 1/3rd of votes (excluding `Abstain` votes) are `NoWithVeto` votes. This means that proposals are accepted is the ratio of `Yes` votes to `No` votes at the end of the voting period is superior to 50% and if the number of `NoWithVeto` votes is inferior to 1/3rd of total votes (excluding `Abstain`).
+
+`Urgent` proposals also work with the aforementioned threshold, except there is another condition that can accelerate the acceptance of the proposal. Namely, if the ratio of `Yes` votes to `InitTotalVotingPower` exceeds 2/3, `UrgentProposal` will be immediately accepted, even if the `Voting period` is not finished. `InitTotalVotingPower` is the total voting power of all bonded Atom holders at the moment when the vote opens.
```

Comment:
I feel like this should be applicable to all proposal. Once a proposal has >2/3 it will always be accepted so we might just end the vote here.

Reply:
No it won't be because of the way we implement it. If validators vote very fast there could be >2/3 yes vote, because they carry the vote of their delegators. When delegators, who will likely be slower to connect and vote cast their tx, they could override the vote of validators and change the outcome. 

It is true that if >2/3rd of validators collude, they can censor votes. However, assuming that validators are honest, it is possible that first validators will vote on an option and then delegators on the other. We don't want to prevent that scenario from happening.

Reply:
oh interesting... okay yeah I'm with gautier on this one... HOWEVER I think we should eliminate the urgent proposal type altogether I've outlined previously

Reply:
I agree with removing the urgent proposal type. It adds confusion and since the rules are then hardwired in the blockchain, it invites people to try to game it.

Leaving one simple and safe on-chain voting mechanism is best. If there are emergencies, they can be handled off-chain, as is currently the case with almost every blockchain.

Reply:
Note that the "one process" rule is the same as urgent.
If you just redefine the "one process" to always act like "regular" (no early end), then there is no problem.
If you really need to go fast and break things, do it off-chain.

Reply:
yup totally in agreement - let's rip it out!